### ISSUE TITLE
UB in shuffle_avx2 and unshuffle_avx2

### ISSUE THREAD

**Issue by EmilDohne** [Blosc/c-blosc2#636]
**Describe the bug**

These two typedefs suggest that both `_src` and `_dest` should be `const uint8_t*`.

```c
typedef void(* shuffle_func)(const int32_t, const int32_t, const uint8_t*, const uint8_t*);
typedef void(* unshuffle_func)(const int32_t, const int32_t, const uint8_t*, const uint8_t*);
```
However, the avx2 implementation does not take a `const uint8_t*` but instead a `uint8_t*`. On further inspection this seems to hold true for other implementations. To my knowledge, all calls to shuffle actually call with a non-const pointer so I think it's likely fine to make these values non-const

**Logs**
Below you can find a `fsanitize=undefined` run that points this out:

https://github.com/EmilDohne/PhotoshopAPI/actions/runs/11365462630/job/31613700651

**Comment by FrancescAlted** [comment]
This sounds reasonable to me.  Would you mind providing a PR?

**Comment by EmilDohne** [comment]
I can definitely have a look at providing one!

**Comment by EmilDohne** [comment]
Fixed by #638
